openmls_rust_crypto/src/provider.rs

```diff
@@ -289,7 +300,12 @@ impl OpenMlsCrypto for RustCrypto {
                 Ok(signature.to_der().to_bytes().into())
             }
             SignatureScheme::ED25519 => {
-                let k = ed25519_dalek::Keypair::from_bytes(key)
+                if key.len() != ed25519_dalek::KEYPAIR_LENGTH {
+                    return Err(CryptoError::CryptoLibraryError);
+                }
+                let mut k = [0u8; ed25519_dalek::KEYPAIR_LENGTH];
```

Comment:
Why do you need to copy it in `k` since you tested before that `pk` has the right length ?